web/src/server/api/routers/media.ts

```diff
@@ -0,0 +1,194 @@
+import { z } from "zod";
+
+import { env } from "@/src/env.mjs";
+import { getMediaStorageServiceClient } from "@/src/features/media/server/getMediaStorageClient";
+import {
+  createTRPCRouter,
+  protectedProjectProcedure,
+} from "@/src/server/api/trpc";
+import { logger } from "@langfuse/shared/src/server";
+import { TRPCError } from "@trpc/server";
+import {
+  MediaContentType,
+  MediaReturnType,
+} from "@/src/features/media/validation";
+
+export const mediaRouter = createTRPCRouter({
+  getById: protectedProjectProcedure
+    .input(z.object({ mediaId: z.string(), projectId: z.string() }))
+    .query(async ({ input, ctx }) => {
+      try {
+        const { projectId, mediaId } = input;
+
+        const media = await ctx.prisma.media.findFirst({
+          where: {
+            projectId,
+            id: mediaId,
+          },
+        });
+
+        if (!media)
+          throw new TRPCError({
+            code: "NOT_FOUND",
+            message: "Media asset not found",
+          });
+        if (!media.uploadHttpStatus)
+          throw new TRPCError({
+            code: "NOT_FOUND",
+            message: "Media not yet uploaded",
+          });
+        if (media.uploadHttpStatus !== 200)
+          throw new TRPCError({
+            code: "NOT_FOUND",
+            message: `Media upload failed`,
+          });
+
+        const mediaStorageClient = getMediaStorageServiceClient(
+          media.bucketName,
+        );
+        const ttlSeconds = env.LANGFUSE_S3_MEDIA_DOWNLOAD_URL_EXPIRY_SECONDS;
+        const urlExpiry = new Date(
+          Date.now() + ttlSeconds * 1000,
+        ).toISOString();
+
+        const url = await mediaStorageClient.getSignedUrl(
+          media.bucketPath,
```

Comment:
what is the structure of bucketPath again? It must never happen that they collide across projects for some reason.